### PR TITLE
refactor: Move compress related code into databend

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

OpenDAL will not maintain those logic.

### CODE REVIEW

**Review by sundy-li — APPROVED** [review]

File: src/common/compress/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,709 @@
...
+use futures::ready;
+use futures::AsyncBufRead;
+use futures::AsyncRead;
+use log::trace;
+use pin_project::pin_project;
+use serde::Deserialize;
+use serde::Serialize;
+
+/// CompressAlgorithm represents all compress algorithm that OpenDAL supports.
+#[derive(Serialize, Deserialize, Copy, Clone, Eq, PartialEq, Debug)]
```
Comment (BohuTANG) [inline src/common/compress/src/lib.rs:46]: It's better to move the logic code from lib.rs to compression.rs

**Review by youngsofun — APPROVED** [review]

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
```
cargo-machete found the following unused dependencies in /workspace:
common-storage -- /workspace/src/common/storage/Cargo.toml:
	async-compression
Done!
```

**Comment by BohuTANG** [comment]
The Dev Linux / check CI failure was solved by https://github.com/datafuselabs/databend/pull/10190

**Pull request merged**